src/auth/cephx/CephxKeyServer.cc

```diff
@@ -436,13 +438,14 @@ int KeyServer::build_session_auth_info(uint32_t service_id,
 				       const AuthTicket& parent_ticket,
 				       CephXSessionAuthInfo& info)
 {
-  if (!get_service_secret(service_id, info.service_secret, info.secret_id)) {
+  double ttl;
+  if (!get_service_secret(service_id, info.service_secret, info.secret_id,
+			  ttl)) {
     return -EACCES;
   }
 
   std::scoped_lock l{lock};
-
-  return _build_session_auth_info(service_id, parent_ticket, info);
+  return _build_session_auth_info(service_id, parent_ticket, info, ttl);
 }
 
 int KeyServer::build_session_auth_info(uint32_t service_id,
```

Comment:
I'm not seeing how callers of this method get the TTL adjustment...

Reply:
cephx layer calls `build_session_auth_info()`, the TTL adjustment happens entirely within:

- `get_service_secret()` returns secret id, secret itself and adjusted TTL
- `_build_session_auth_info()` takes adjusted TTL and sets ticket timestamps and validity accordingly

The result is `CephXSessionAuthInfo` (passed as `info`, by non-const reference) with the filled-in ticket.

Reply:
What about this second ``build_session_auth_info`` though?  It's calling ``_build_session_auth_info()`` and passing in the config value for the TTL

Reply:
The second one is different.  It is called by the monitor with the mon secret (i.e. secret_id == -1).  The monitor supplies the secret and `info.validity` isn't ever looked at.  I kept `cct->_conf->auth_service_ticket_ttl` there just in case.

See `Monitor::get_authorizer()`.

Reply:
Hmm, I can see how the signature of the second `build_session_auth_info()` is confusing -- `service_secret` and `secret_id` are input parameters.  I'll update it.

Reply:
Oh I see, thanks!  

We should consider updating all of these methods to use pointers (instead of references) for output args in a later cleanup.

Reply:
Cleaned up the second overload, with references for now to fit the rest of that class.